lib/stripe/card_cloner.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+# Here we clone
+# - a card (card_*) stored in a customer in a platform account
+# into
+# - a payment method (pm_*) in a new customer in a connected account
+#
+# This process is used in the migration between using the Stripe Charges API (stripe_connect)
+#   and the Stripe Payment Intents API (stripe_sca)
+#
+# This process can be deleted once all hubs are running on the new stripe_sca method and all cards in the system have been migrated to the new payment_methods
+#   Basically, when all DBs have no card_* values in credit_card.gateway_payment_profile_id
+module Stripe
+  class CardCloner
+    def clone!(credit_card, stripe_account_id)
+      return unless credit_card.gateway_payment_profile_id.starts_with?('card_')
+
+      new_payment_method = clone_payment_method(credit_card, stripe_account_id)
+      new_customer = Stripe::Customer.create({ email: credit_card.user.email },
+                                             stripe_account: stripe_account_id)
+      attach_payment_method_to_customer(new_payment_method.id, new_customer.id, stripe_account_id)
+
+      credit_card.update_attributes gateway_customer_profile_id: new_customer.id,
+                                    gateway_payment_profile_id: new_payment_method.id
+      credit_card
```

Comment:
You don't need a return value here, do you?

Once we converted a card to the new API, we can't switch back to Stripe Connect, can we?

Reply:
this was changed in later commits.

This process clones a card in a platform account to the connected account.
This process is backward compatible! Because the original stripe method will take the card in a platform account and create a "token" in the connected account, this method takes the card in the platform account and create another card in the connected account.